File: src/RenderEngine.Core/Mathf.cs
using System;

namespace RenderEngine.Core
{
    public static class Mathf
    {
        public const float Epsilon = 0.00001f;

        public static bool Approximately(float a, float b)
        {
            return Math.Abs(a - b) < Epsilon;
        }
    }
}

File: lib/ray-tracer-core/Mathf.cpp
#include <math.h>

#include "Mathf.h"

const double Mathf::Epsilon = 0.00001f;

bool Mathf::Approximately(float a, float b)
{
	return fabs((double)a - b) < Epsilon;
}

bool Mathf::Approximately(double a, double b)
{
	return fabs((double)a - b) < Epsilon;
}

double Mathf::ToZeroIfClose(double a)
{
	return Approximately(a, 0) ? 0 : a;
}
